### ISSUE TITLE
ZoneSum vs VentilationRateProcedure when one zone has zero oa requirement

### ISSUE THREAD

**Issue by BobHenninger** [NREL/EnergyPlus#4399]
Helpdesk Ticket 9523

Log of User and MJW comments

Looking at the table outputs.

 REPORT: HVAC Sizing Summary
 FOR: Entire Facility
 Zone Cooling 

 This report shows the same flow rates for all zones in both runs.

 But this report starts to show differences (look at the last value, User Heating):

 System Design Air Flow Rates 

 Calculated cooling [m3/s] User cooling [m3/s] Calculated heating [m3/s] User heating [m3/s] 
 v0 DOAS 2.98 0.74 0.74 0.74

 v1 DOAS 2.98 2.98 0.74 2.98 

 Then we get to REPORT: Component Sizing Summary

 AirLoopHVAC 

 Design Supply Air Flow Rate [m3/s]
 v0 DOAS 0.740822
 v1 DOAS 2.98

 Maximum Outdoor Air Flow Rate [m3/s] 
 v0 DOAS OA CONTROLLER 0.740822 
 v1 DOAS OA CONTROLLER 2.98

 So, the VRP DOAS is bumping up the system flow rate by 4.022x. Why?

 Looking further at the Zone Ventilation Parameters report, I see several zones (toilets and kitchen) that have zero OA requirements. I suspect that these are impacting the VRP method somehow because the total floor area comes into play.

 In the Zone Ventilation Calculations for Cooling Design report, there are some zones that show Primary Outdoor Air Fraction - Zpz <1.0. Not sure why some show that and not others. But as a next step, I'd suggest entering a tiny OA flow requirement for the zones that are currently zero, or remove them from the DOAS system. Then see if the two methods result in the same total DOAS system and OA design flow rates. 

 If they do, then we need to pursue whether the treatment of the zero-OA zones is being handled properly within the VRP calcs when using VentilationRequirement to size a system. My guess is if you changed the sizing:system Type of Load from VentilationRequirement to Sensible, then you would also get the same OA flow and system flow between the two runs. I think it's the combination of VentilationRequirement, some zones with zero OA flow, and VRP that is causing a problem.

 Mike

 > Mike, thanks for your comment. That is exactly my point: if the
 > effectivenesses is 1.0 and max zone oa fraction is 1.0 then both
 > method should have the same result. I did look at the Tabular report
 > (see attached). The sizing did have different ventilation flow rates.
 > However, I cannot replicate the flowrate with manual calculation
 > because the design specifications are identical in both versions.
 > 
 > Also attached is the OA flowrate schedule in CSV file.
 > 
 > Any further insight is appreciated. Note that I NEED to specify OA
 > sizing method as VRP, because the 62.1 summary table (in Tabular
 > report) will only be populated if VRP method is selected. It makes my
 > life easier on the reporting.
 > 
 > Thanks,
 > 
 > Ery

User files
Schedule_Minimum_OA_v2.csv.zip
HM_Proposed_RC_Tabular_Report_Comparison.xls
QC_HM_Proposed_RC_OA_Comparison.zip

**Comment by rraustad** [comment]
Why does this field have a default? If it is a cooling only radiant system this field is not needed and could be left blank. 

ZoneHVAC:LowTemperatureRadiant:VariableFlow,
   A6 , \field Heating Design Capacity Method
        \required-field
        \type choice
        \key HeatingDesignCapacity
        \key CapacityPerFloorArea
        \key FractionOfAutosizedHeatingCapacity
        \default HeatingDesignCapacity

This trips an error:

   *\* Severe  *\* ZoneHVAC:LowTemperatureRadiant:VariableFlow = FL1 CAFE RADIANT FLOOR
   *\*   ~~~   *\* Input for Heating Design Capacity Method = HEATINGDESIGNCAPACITY
   *\*   ~~~   *\* Blank field not allowed for Heating Design Capacity

Workaround: Enter a value for Heating Design Capacity (1.0 works).

**Comment by mjwitte** [comment]
@rraustad That's a reasonable default with Heating Design Capacity defaulting to autosize.  It appears that way in other places.  Looks like getinput for this is broken.

**Comment by rraustad** [comment]
I can probably fix that by looking for HW nodes and only error out if those nodes are blank. What other place so I can see what they did there and if those places require a heating component.

Regarding this issue, what should I do here? The HVAC Sizing Summary air flows are based on the zone load and have nothing to do with OA. So both files should be the same.

PreDefTableEntry( pdchZnClCalcDesAirFlow, curName, CalcFinalZoneSizing( CtrlZoneNum ).DesCoolVolFlow, 3 );

As described, the air loop flow did change. I guess I'm not psyc'd to look through the mech vent calc's to see if they work correctly.

**Comment by rraustad** [comment]
Oh, I see the user thinks the air loop flows should be the same. I'll look a little more.

**Comment by rraustad** [comment]
Looks like all the other radiant that use "default HeatingDesignCapacity" are heating only systems. So this object will need a little more logic.

**Comment by rraustad** [comment]
I printed out the ventilation report to see where differences lie. VRP uses the VAV minimum to callculate OA. I have to check to see if ZoneSum also does this.

Should this always be reported (for ZoneSum and VentilationRateProcedure)?

First half of figure is ZoneSum Cooling and Heating. VentilationRateProcedure results are next.

![zonesum_result_prefix](https://cloud.githubusercontent.com/assets/5778156/9966585/b3c1fbde-5e0b-11e5-8b06-e528ddcfcb2c.png)

**Comment by mjwitte** [comment]
@rraustad It may be instructive to look at the same tables for the workaround case where a tiny OA requirement is added to the zero OA zones.

**Comment by rraustad** [comment]
Ah, this is the variable used to calc OA in the VRP procedure:

// In case for some reason the VAV minimum has not been defined, use the design primary airflow

ZonePA = FinalZoneSizing( CtrlZoneNum ).DesCoolVolFlow;
// Save VpzMin in case this is a single path VAV system.
// Std 62.1-2010, section 6.2.5.1: "For VAV-system design purposes, Vpz is the lowest zone primary
// airflow value expected at the design condition analyzed."
VpzMinClgByZone( CtrlZoneNum ) = FinalZoneSizing( CtrlZoneNum ).DesCoolVolFlowMin;

if ( FinalZoneSizing( CtrlZoneNum ).DesCoolVolFlowMin <= 0 ) VpzMinClgByZone( CtrlZoneNum ) = ZonePA;

if ( FinalZoneSizing( CtrlZoneNum ).ZoneSecondaryRecirculation > 0.0 || FinalZoneSizing( CtrlZoneNum ).DesCoolVolFlowMin <= 0 ) {
    // multi-path system or VAV Minimum not defined
    ZoneOAFracCooling = FinalZoneSizing( CtrlZoneNum ).VozClgByZone / FinalZoneSizing( CtrlZoneNum ).DesCoolVolFlow;
} else {
    // Single path; Use VAV Minimum as the Vpz in the Zp = Voz / Vpz equations
    ZoneOAFracCooling = FinalZoneSizing( CtrlZoneNum ).VozClgByZone / FinalZoneSizing( CtrlZoneNum ).DesCoolVolFlowMin;
}

So for zones with 0 OA, the supply air flow rate is used instead of 0. Why can't a zone have a 0 OA fraction?

**Comment by mjwitte** [comment]
So, those zones also have a zero min flow?  So maybe it's not the zero OA that's the problem, but the zero min flow.  Which, if you have OA would get bumped up by the OA requirement, maybe?

**Comment by rraustad** [comment]
Changing OA > 0 did not change the answer (tables same as above). I think it's the fact that the VAV minimum flow is used in VRP and not in ZoneSum. I'm not sure if that's correct or not. If OA is specified for ZoneSum, should E+ also look at supply air flow (i.e., OAfrac) to ensure that amount of OA is provided to the zone?

```
DesignSpecification:OutdoorAir,
 SZ DSOA THERMAL ZONE: FLM TOILET,  !- Name
 sum,                     !- Outdoor Air Method
 0.0000,                  !- Outdoor Air Flow per Person {m3/s-person}
 0.0000,                  !- Outdoor Air Flow per Zone Floor Area {m3/s-m2}
 0.0000001;               !- Outdoor Air Flow per Zone {m3/s}
```

**Comment by mjwitte** [comment]
So much for that hypothesis, although 0.0000001 may fall below some smallflow threshold, might try something bigger.  Also, did you notice the comments about VentilationRequirement in sizing vs SensibleLoad?

**Comment by rraustad** [comment]
VRP does this while ZoneSum does not:

ZoneOAFracCooling = FinalZoneSizing( CtrlZoneNum ).VozClgByZone / FinalZoneSizing( CtrlZoneNum ).DesCoolVolFlowMin;

ClgSupplyAirAdjustFactor = ZoneOAFracCooling / FinalSysSizing( AirLoopNum 
).MaxZoneOAFraction;

FinalZoneSizing( CtrlZoneNum ).DesCoolVolFlowMin *= ClgSupplyAirAdjustFactor;

So I think the ZoneSum result will always be different than VentilationRateProcedure unless ZoneSum accounts for OAfrac (i.e., the supply flow that allows the specified OA to be provided at the minimum VAV stop).

**Comment by mjwitte** [comment]
ZoneSum shouldn't be changed, it's correct.  So, I went searching on VozClgByZone  and found this code

```
                            // CR 8872 - check to see if uncorrected OA is calculated to be greater than 0
                            if ( ! ( ZoneOAUnc > 0.0 ) ) {
                                ShowSevereError( "Sizing:System - The system outdoor air method is set to VRP in " + FinalSysSizing( AirLoopNum ).AirPriLoopName );
                                ShowContinueError( "But zone \"" + FinalZoneSizing( CtrlZoneNum ).ZoneName + "\" associated with system does not have OA flow/person" );
                                ShowContinueError( "or flow/area values specified in DesignSpecification:OutdoorAir object associated with the zone" );
                            }
```

Looks like the same issue came up before in #3645.  Maybe VRP is simply undefined for this case and the sever should be fatal?  To answer that we need to consult the ASHRAE standard for VRP methodology.

**Comment by rraustad** [comment]
"notice the comments about VentilationRequirement in sizing vs SensibleLoad?"

where do I look for this?

**Comment by mjwitte** [comment]
That's up at the top of this page in my responses to the user 
"If they do, then we need to pursue whether the treatment of the zero-OA zones is being handled properly within the VRP calcs when using VentilationRequirement to size a system. My guess is if you changed the sizing:system Type of Load from VentilationRequirement to Sensible, then you would also get the same OA flow and system flow between the two runs. I think it's the combination of VentilationRequirement, some zones with zero OA flow, and VRP that is causing a problem."

But that may just be a diversion.

**Comment by rraustad** [comment]
I believe it's not the zones with 0 OA that is causing the SA to be higher for the VRP case, it's the zones with OA >0 and minimum VAV stops < 1 causing this. I'll try to set up a test case where minimum VAV stop = 1 for non-zero OA zones only.

**Comment by rraustad** [comment]
Hmm, same answer.

**Comment by rraustad** [comment]
In order to get the same answer with ZoneSum and VRP, the zone OA fraction must = the system OA fraction to ensure the system cooling/heating effectiveness = 1. If a zone's OA fraction is less than the system OA fraction, that means there is one or more zones with zone OA fraction > system OA fraction which would force SysCoolingEV < 1. So to make these 2 simulations equal, the zone OA fraction must equal the system OA fraction in all zones. That leads to using only an OA flow per zone floor area input in the DesignSpecification:OutdoorAir object (which is unlikely given requirements for different zone types)..

SysCoolingEv = System Ventilation Efficiency - Ev in the System Ventilation Requirements report.

SysCoolingEv = 1.0 + Xs - ZoneOAFrac; (for single-path ventilation systems)

if ( SysCoolingEv < MinCoolingEvz ) MinCoolingEvz = SysCoolingEv;

Xs = min( 1.0, FinalSysSizing( AirLoopNum ).SysUncOA / SysSizing( CurOverallSimDay, AirLoopNum ).DesCoolVolFlow );

ZoneOAFrac = FinalZoneSizing( CtrlZoneNum ).ZpzClgByZone;

FinalZoneSizing( CtrlZoneNum ).ZpzClgByZone = min( 1.0, FinalZoneSizing( CtrlZoneNum ).VozClgByZone / VpzMinClgByZone( CtrlZoneNum ) );

FinalZoneSizing( CtrlZoneNum ).VozClgByZone = ZoneOAUnc / FinalZoneSizing( CtrlZoneNum ).ZoneADEffCooling;

VpzMinClgByZone( CtrlZoneNum ) = FinalZoneSizing( CtrlZoneNum ).DesCoolVolFlowMin;

FinalZoneSizing( CtrlZoneNum ).DesCoolVolFlowMin = max( FinalZoneSizing( CtrlZoneNum ).DesCoolMinAirFlow, FinalZoneSizing( CtrlZoneNum ).DesCoolMinAirFlow2, FinalZoneSizing( CtrlZoneNum ).DesCoolVolFlow \* FinalZoneSizing( CtrlZoneNum ).DesCoolMinAirFlowFrac );

Setting the zone sizing minimum airflow fraction = 1 got the answers much closer but not the same. Still looking.

```
Sizing:Zone,
 1,                       !- Cooling Minimum Air Flow Fraction
```

This run also has minimum VAV stop = 1 for all zones (not sure if that's needed or not).

```
AirTerminal:SingleDuct:VAV:NoReheat,
  1.0,                     !- Constant Minimum Air Flow Fraction
```

![vrp_closer](https://cloud.githubusercontent.com/assets/5778156/9977898/119f08ac-5ee6-11e5-9d6d-a757fa9ff6d1.png)

**Comment by rraustad** [comment]
Changing VAV min back to 0 did not change the result above.

```
AirTerminal:SingleDuct:VAV:NoReheat,
 0.0,                     !- Constant Minimum Air Flow Fraction
```

**Comment by rraustad** [comment]
The 62.1 Summary has a column named "Percent Outdoor Air - %OA" which shows a fraction (see figure above). Should I change this to be %?

**Comment by rraustad** [comment]
Fixed value in % column. Fixed ADS Eff reporting for ZoneSum method.

Results for ZoneSum method.

All zones changed to 0.8 ADS Eff:

```
DesignSpecification:ZoneAirDistribution,
  SZ DSZAD THERMAL ZONE: FL1 CAFE,  !- Name
  0.8,   !- Zone Air Distribution Effectiveness in Cooling Mode {dimensionless}
  0.8;   !- Zone Air Distribution Effectiveness in Heating Mode {dimensionless}
```

All zone have 0 minimum air flow fraction and 0 VAV minimum stop:

```
Sizing:Zone,
  0,                       !- Cooling Minimum Air Flow Fraction

AirTerminal:SingleDuct:VAV:NoReheat,
  0.0,                     !- Constant Minimum Air Flow Fraction
```

![zonesum_point8adseff](https://cloud.githubusercontent.com/assets/5778156/9984451/99f8a154-5fe5-11e5-9c06-70968b53ffea.png)

**Comment by mjwitte** [comment]
Closed via #5216
